webapp/channels/src/components/channel_threads_rhs/header.tsx

```diff
@@ -0,0 +1,155 @@
+                defaultMessage='Collapse the right sidebar'
+            />
+            <KeyboardShortcutSequence
+                shortcut={KEYBOARD_SHORTCUTS.navExpandSidebar}
+                hideDescription={true}
+                isInsideTooltip={true}
+            />
+        </Tooltip>
+    );
+
+    return (
+        <div className='sidebar--right__header'>
+            <span className='sidebar--right__title'>
+
+                {canGoBack && (
+                    <BackButton
+                        className='sidebar--right__back'
+                        onClick={goBack}
+                    >
+                        <i
+                            className='icon icon-arrow-back-ios'
+                            aria-label='Back Icon'
+                        />
+                    </BackButton>
+                )}
+
+                <HeaderTitle>
+                    <FormattedMessage
+                        id='channel_threads_rhs.header.title'
+                        defaultMessage='All Threads'
+                    />
+                </HeaderTitle>
+
+                {channel.display_name &&
+                    <span
+                        className='style--none sidebar--right__title__subtitle'
+                    >
+                        {channel.display_name}
+                    </span>
+                }
+            </span>
+
+            <OverlayTrigger
+                delayShow={Constants.OVERLAY_TIME_DELAY}
+                placement='bottom'
+                overlay={isExpanded ? shrinkSidebarTooltip : expandSidebarTooltip}
+            >
+                <button
+                    type='button'
+                    className='sidebar--right__expand btn-icon'
+                    onClick={toggleRhsExpanded}
+                >
+                    <LocalizedIcon
+                        className='icon icon-arrow-expand'
+                        ariaLabel={{id: t('rhs_header.expandSidebarTooltip.icon'), defaultMessage: 'Expand Sidebar Icon'}}
+                    />
+                    <LocalizedIcon
+                        className='icon icon-arrow-collapse'
+                        ariaLabel={{id: t('rhs_header.collapseSidebarTooltip.icon'), defaultMessage: 'Collapse Sidebar Icon'}}
+                    />
+                </button>
+            </OverlayTrigger>
+
+            <OverlayTrigger
+                delayShow={Constants.OVERLAY_TIME_DELAY}
+                placement='top'
+                overlay={closeSidebarTooltip}
+            >
+                <button
+                    id='rhsCloseButton'
+                    type='button'
+                    className='sidebar--right__close btn-icon'
+                    aria-label='Close'
+                    onClick={onClose}
+                >
+                    <LocalizedIcon
+                        className='icon icon-close'
+                        ariaLabel={{id: t('rhs_header.closeTooltip.icon'), defaultMessage: 'Close Sidebar Icon'}}
```

Comment:
@calebroseland from MDN: 
> By default, a button's accessible name is the content between the opening and closing [button](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button) tags

I'm not 100% sure but it seems that it will pick up the child's `aria-label` (if the button has none), and will use that as an accessible name.
Accessibility inspection in both Firefox and Chrome display it correctly.

| chrome | firefox |
|--------|--------|
|<img width="488" alt="Screenshot 2023-04-26 at 2 24 22 PM" src="https://user-images.githubusercontent.com/3829551/234560794-e7e231a5-c5ee-4acc-8724-de8718385509.png">|<img width="553" alt="Screenshot 2023-04-26 at 2 25 06 PM" src="https://user-images.githubusercontent.com/3829551/234560917-a3c809a3-ad80-43c3-9df1-a111156caefd.png">|

I see that we use that in all RHS views as well.
The reason might be that we handle the expand/collapse by CSS so JS doesn't know, 
making it difficult to add the text as `aria-label` on the button.